tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/ServerlessMiniAgent.cs

```diff
@@ -0,0 +1,113 @@
+    internal const string GCPFunctionDeprecatedEnvVarIdentifier = "GCP_PROJECT";
+    internal const string GCPFunctionNewerNameEnvVar = "K_SERVICE";
+    internal const string GCPFunctionNewerEnvVarIdentifier = "FUNCTION_TARGET";
+
+    internal static bool IsGCPFunction { get; private set; } = GetIsGCPFunction();
+
+    internal static bool IsAzureFunction { get; private set; } = GetIsAzureFunction();
+
+    internal static void MaybeStartMiniAgent(ServerlessMiniAgentManager manager)
+    {
+        if (!IsGCPFunction && !IsAzureFunction)
+        {
+            return;
+        }
+
+        Log.Information("Trying to start the mini agent");
+
+        string rustBinaryPath;
+        if (Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH") != null)
+        {
+            rustBinaryPath = Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH");
+        }
+        else
+        {
+            // Environment.OSVersion.Platform can return PlatformID.Unix on MacOS, this is OK as GCP & Azure don't have MacOs functions.
+            if (Environment.OSVersion.Platform != PlatformID.Unix && Environment.OSVersion.Platform != PlatformID.Win32NT)
+            {
+                Log.Error("Serverless Mini Agent is only supported on Windows and Linux.");
+                return;
+            }
+
+            var dirPathSep = Path.DirectorySeparatorChar;
+            var isWindows = Environment.OSVersion.Platform == PlatformID.Win32NT;
+
+            string rustBinaryPathRoot;
+            if (IsGCPFunction)
+            {
+                rustBinaryPathRoot = "/layers/google.dotnet.publish/publish/bin";
+            }
+            else
+            {
+                // IsAzureFunction
+                if (isWindows)
+                {
+                    rustBinaryPathRoot = "C:\\home\\site\\wwwroot";
+                }
+                else
+                {
+                    // linux
+                    rustBinaryPathRoot = "/home/site/wwwroot";
+                }
+            }
+
+            string rustBinaryPathOsFolder = isWindows ? "datadog-serverless-agent-windows-amd64" : "datadog-serverless-agent-linux-amd64";
+            rustBinaryPath = string.Format("{0}{1}{2}{3}datadog-serverless-trace-mini-agent", rustBinaryPathRoot, dirPathSep, rustBinaryPathOsFolder, dirPathSep);
+        }
+
+        manager.Start(rustBinaryPath);
+    }
+
+    internal static bool GetIsGCPFunction()
+    {
+        bool isDeprecatedGCPFunction = Environment.GetEnvironmentVariable(GCPFunctionDeprecatedNameEnvVar) != null && Environment.GetEnvironmentVariable(GCPFunctionDeprecatedEnvVarIdentifier) != null;
+        bool isNewerGCPFunction = Environment.GetEnvironmentVariable(GCPFunctionNewerNameEnvVar) != null && Environment.GetEnvironmentVariable(GCPFunctionNewerEnvVarIdentifier) != null;
+
+        return isDeprecatedGCPFunction || isNewerGCPFunction;
+    }
+
+    internal static bool GetIsAzureFunction()
```

Comment:
Already exists in https://github.com/DataDog/dd-trace-dotnet/blob/fdd89ed264bc8aeb33d9a9c6c14fbb1ada1701d0/tracer/src/Datadog.Trace/Configuration/ImmutableAzureAppServiceSettings.cs#L116

Reply:
We need to call `GetIsAzureFunction` in a couple places before `ImmutableAzureAppServiceSettings` is initialized, so I opted to keep it as a static function

Reply:
Ok nitpicking and clearly debatbke but I would have placed all the code in the same place then (in ImmutableSettings). So this static function could be there, and you could use it to fill the ImmutableSettings upon construction to avoid code duplucation